src/Compilers/Core/Portable/CodeGen/ILBuilderConversions.cs

```diff
@@ -179,6 +183,11 @@ public void EmitNumericConversion(Microsoft.Cci.PrimitiveTypeCode fromPredefType
                             else
                                 this.EmitOpCode(ILOpCode.Conv_i); // potentially widening, so not NOP
                             break;
+                        case Microsoft.Cci.PrimitiveTypeCode.Pointer:
+                        case Microsoft.Cci.PrimitiveTypeCode.FunctionPointer:
+                            if (@checked)
+                                goto default;
```

Comment:
>goto default; [](start = 32, length = 13)

Is this necessary? It looks like UIntPtr, Pointer and FunctionPointer are unsigned types of the same size.